review/fileToReview.js

```diff
@@ -12,62 +17,42 @@ var domain = "bank.local.fr"
-        "jws": jws,
-        "Content-type": "application/json",
-        "Accept": "application/json"
-      }
-    } else {
-      headers = {
-        "Authorisation": authorization,
-        "Content-type": "application/json",
-        "Accept": "application/json",
-      }
-    }
+  const headers = jws ? {
+    ...config.defaultHeaders,
+    "jws": jws,
+    "Authorisation": authorization
+  } : {...config.defaultHeaders, "Authorisation": authorization};
+
+  const {code, response} = await doRequest('GET', `${config.domain}/accounts/${id}/transactions?page=${page}`, headers);
 
-	  var {code, response } = await doRequest('GET',
-      domain + '/accounts/'+ id + '/transactions?' + `page=${page}`,
-      headers);
+  if (code < 200 || code >= 400 || !response?.data) {
+    console.error(`Bad request received from bankin API : ${code}`, response);
+    return [];
+  }
+
+  if (response?.data?.meta && response?.data?.meta?.hasPageSuivante) {
+    const mouvements = response.data.Mouvements;
+    const date = mouvements[mouvements.length - 1].dateValeur;
+
+    if (date <= fromDate) {
+      console.log("FromDate is Reached - we don't need more transaction");
+    } else {
 
+      if (!mouvements)
+        throw new Error("Empty list of transactions ! " + JSON.stringify(previousTransactions));
 
-		if (response && code == 200 && response.data) {
-      if (response.data.meta) {
-        if (response.data.meta.hasPageSuivante) {
-          let mouvements = response.data.Mouvements;
-          var date = mouvements[mouvements.length -1].dateValeur;
-          if (date <= fromDate) {
-            console.log("FromDate is Reached - we don't need more transaction");
-          } else {
-            // if we have mouvements
-            if (mouvements) {
-              if (assertTransactions(mouvements)) {
-                return [];
-              } else {
-                console.log(`Push transactions from page ${page}`);
-              }
-            } else {
-              throw new Error("Empty list of transactions ! " + JSON.stringify(previousTransactions));
-            }
-            let nextPagesTransactions = fetchTransactions(fromDate, authorization, (jws || null), id, page + 1, mouvements);
-            response.data.Mouvements = mouvements.concat(nextPagesTransactions);
-          }
-        }
+      if (assertTransactions(mouvements)) {
+        return [];
+      } else {
+        console.log(`Push transactions from page ${page}`);
       }
-      return response.data.Mouvements;
-    } else throw new Error();
 
-    return [];
-	} catch (err) {
-		throw new CustomError({
-      function: 'fetchTransactions',
-			statusCode: 'CRASH',
-			rawError: e,
-		});
-	}
+      const nextPagesTransactions = fetchTransactions(fromDate, authorization, (jws || null), id, page + 1, mouvements);
+      response.data.Mouvements = mouvements.concat(nextPagesTransactions);
+    }
+  }
+
+  return response.data.Mouvements;
```

Comment:
You are fetching the WHOLE list of movements before returning them. This is to avoid in case of lot of data. 
It's a better practice the process by batch to avoid a huge allocated memory.